### PR TITLE
Add reference to `QMCSampler` in tutorial

### PR DESCRIPTION
**Pull request by himkt**
This PR adds a reference to `QMCSampler` in the tutorial that was introduced in #2423.

### CODE REVIEW

**Comment by himkt** [comment]
@HideakiImamura @nzw0301 Could you please take a look?

File: tutorial/10_key_features/003_efficient_optimization_algorithms.py [hunk 1]
```diff
@@ -90,7 +92,7 @@
 # :func:`~optuna.trial.Trial.should_prune` decides termination of the trial that does not meet a predefined condition.
 #
 # We would recommend using integration modules for major machine learning frameworks.
-# Exclusive list is :mod:`optuna.integration` and usecases are available in  `optuna/examples <https://github.com/optuna/optuna-examples/>`_.
+# Exclusive list is :mod:`~optuna.integration` and usecases are available in  `~optuna/examples <https://github.com/optuna/optuna-examples/>`_.
```
[the remark this answers is not in the document]
Comment (himkt) [reply tutorial/10_key_features/003_efficient_optimization_algorithms.py:95]: Thank you. 🙇‍♂️

**Review by nzw0301 — APPROVED** [review]
LGTM

File: tutorial/10_key_features/003_efficient_optimization_algorithms.py [hunk 2]
```diff
@@ -64,23 +66,23 @@
 #
 # Optuna provides the following pruning algorithms:
 #
-# - Median pruning algorithm implemented in :class:`optuna.pruners.MedianPruner`
+# - Median pruning algorithm implemented in :class:`~optuna.pruners.MedianPruner`
 #
-# - Non-pruning algorithm implementd in :class:`optuna.pruners.NopPruner`
```
Comment (himkt) [inline tutorial/10_key_features/003_efficient_optimization_algorithms.py:69]: tweak typo fix.

**Comment by himkt** [comment]
Let me kindly ping @HideakiImamura. :pray:

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**